### DISCUSSION TITLE
State management by reactivity?

### DISCUSSION THREAD

**Discussion by jgarte** [yukikurage/purescript-jelly#41] [Q&A] [answered]
Hi, What does state management by reactivity mean? 

I'm `a :: Newb`

**Comment by yukikurage** [comment, marked as answer]
Hello. Thank you for your interest in Jelly.

Jelly uses reactivity in very limited places, not throughout the app.

That is component state management.

The `Signal a` type is a proprietary implementation, but it is similar to `purescript-signal`, i.e., it represents a changing value, yet it can detect the change and execute an Effect.

Jelly uses the `Signal a` type to detect state changes and rewrite the DOM directly. I call this procedure reactivity.

**Reply by yukikurage** [reply]
Some say that such limited reactivity is RP (Reactive Programming), not FRP (Functional Reactive Programming). I take that into account and assume that this library is RP.
